packages/std/extra/live_update_from_npm_packages.bri

```diff
@@ -0,0 +1,144 @@
+import * as std from "/core";
+import { withRunnable } from "./runnable.bri";
+import { DEFAULT_LIVE_UDPATE_REGEX_VERSION_MATCH } from "./live_update.bri";
+import type {} from "nushell";
+
+// HACK: The `import type` line above is a workaround for this issue:
+// https://github.com/brioche-dev/brioche/issues/242
+
+/**
+ * Extra options for the live update from NPM packages.
+ *
+ * @param packageName - The name of the NPM package to update.
+ */
+interface LiveUpdateFromNpmPackagesExtraOptions {
+  packageName: string;
+}
+
+/**
+ * Options for the live update from NPM packages.
+ *
+ * @param project - The project export that should be updated. Must include a
+ *   `extra.packageName` property containing the name of the NPM package.
+ * @param matchVersion - A regex value (`/.../`) to extract the version number from
+ *   a string. The regex must include a group named "version". If not
+ *   provided, an optional "v" prefix will be stripped and the rest of the
+ *   version will be checked if it's a semver or semver-like version number.
+ */
+interface LiveUpdateFromNpmPackagesOptions {
+  project: { version: string; extra: LiveUpdateFromNpmPackagesExtraOptions };
+  matchVersion?: RegExp;
```

Comment:
...any idea how useful this option will be? I don't think I've ever seen an NPM package that didn't use an `x.y.z` style version number (unlike git tags, where we already have several packages with weird tag names). I also haven't looked into NPM's rules to see what they allow for versions, I might try and do some research on this as a follow-up....

Reply:
There is no need indeed to have the `matchVersion` field here if SemVer is respected.